src/Controller/TaskController.php

```diff
@@ -62,30 +62,26 @@ public function editAction(Task $task, Request $request)
         ]);
     }
 
-    /**
-     * @Route("/tasks/{id}/toggle", name="task_toggle")
-     */
-    public function toggleTaskAction(Task $task)
+    #[Route('/tasks/{id}/toggle', name: 'app_task_toggle')]
+    public function toggleTaskAction(ManagerRegistry $doctrine, Task $task): RedirectResponse
     {
         $task->toggle(!$task->isDone());
-        $this->getDoctrine()->getManager()->flush();
+        $doctrine->getManager()->flush();
 
         $this->addFlash('success', sprintf('La tâche %s a bien été marquée comme faite.', $task->getTitle()));
 
-        return $this->redirectToRoute('task_list');
+        return $this->redirectToRoute('app_task_list');
     }
 
-    /**
-     * @Route("/tasks/{id}/delete", name="task_delete")
-     */
-    public function deleteTaskAction(Task $task)
+    #[Route('/tasks/{id}/delete', name: 'app_task_delete')]
+    public function deleteTaskAction(ManagerRegistry $doctrine, Task $task): RedirectResponse
     {
-        $em = $this->getDoctrine()->getManager();
+        $em = $doctrine->getManager();
```

Comment:
tu peux la variable $em (entity manager) en variable de class, après tu peux faire $this->em;

Reply:
Fixed => See commit [Issue #4 Implementation of automated tests](https://github.com/l-lemaitre/todolist/pull/9/commits/cddb13e1ca46e6ffbaba7587e68c926c7aedbaa7)